### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by bfbachmann**
## Related Issue
#74

## Description
* Add tests for React components and modules on the front end
* Add Travis config so tests are run and coverage is reported

## WIKI Updates
None

## Todos

General:
- [X] Tests
- [X] Documentation
- [ ] Wiki

Other (links to TODOs in code):
* None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,16 @@
 # Sleuth Frontend :computer:
+[![ZenHub](https://img.shields.io/badge/Shipping_faster_with-ZenHub-5e60ba.svg?style=flat)](https://zenhub.com)
```
Comment (bobheadxi) [inline README.md:2]: crisp

File: src/components/SearchForm.js [hunk 2]
```diff
@@ -107,7 +106,6 @@ export default class SearchForm extends React.Component {
         this.setState({
             results: results,
             noResults: results.length === 0,
-            queryId: this.state.queryId + 1,
```
Comment (bobheadxi) [inline src/components/SearchForm.js:110]: What happened here? I thought this was needed to get the graph view refreshing, did you work out a way to do it without?

File: test/components/ResultItem.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        const desc = div.childAt(3);
+        expect(desc.type()).toEqual('p');
+        expect(desc.text()).toEqual('there is something amazing about code');
+        expect(desc.childAt(0).text()).toEqual('amazing');
+        expect(desc.childAt(0).type()).toEqual('b');
+        expect(desc.childAt(1).text()).toEqual('code');
+        expect(desc.childAt(1).type()).toEqual('b');
+    });
+
+    it('should render correctly without highlights', () => {
```
Comment (bobheadxi) [inline test/components/ResultItem.test.js:39]: 👍

File: test/components/ResultTypes.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    GenericPageResult
+} from './../../src/components/ResultTypes';
+
+const id = 'TEST_ID';
+const description = 'TEST_DESCRIPTION';
+const pageName = 'TEST_PAGENAME';
+const siteName = 'TEST_SITENAME';
+const links = ['TEST_LINK1', 'TEST_LINK2'];
+
+describe('CourseItemResult', () => {
```
Comment (bobheadxi) [inline test/components/ResultTypes.test.js:14]: just wondering, is it standard practice to unit test objects like this that have no special functions (ie these result types that exist solely to hold data fields)?

**Review by bobheadxi — APPROVED** [review]
2clean

File: README.md (hunk 1 above)
Comment (bfbachmann) [reply README.md:2]: We're definitely doing it big out here

File: test/components/ResultTypes.test.js (hunk 4 above)
Comment (bfbachmann) [reply test/components/ResultTypes.test.js:14]: Nope! Haha. This really doesn't need to be tested, I just wanted good coverage numbers ;)

File: src/components/SearchForm.js (hunk 2 above)
Comment (bfbachmann) [reply src/components/SearchForm.js:110]: This is just to prevent us from re-creating the whole graph if the query the user entered hasn't changed.

**Pull request merged**